WordPress/gutenberg
# DataViews: introduce locked view filters

Alternative to https://github.com/WordPress/gutenberg/pull/62157
Closes https://github.com/WordPress/gutenberg/issues/60472
Partially addresses https://github.com/WordPress/gutenberg/issues/60468

## What?

Enables consumers of DataViews to declare view filters that are locked: they cannot be edited or removed by users — and they are also hidden.

## Why?

There are situations in which the view 1) wants to communicate a filter state so consumers can filter the datasource but 2) don't want that filter to be exposed to users.

## How?

Introduces the `isLocked` prop to view filters: a filter cannot be edited or removed by the user when it's `true`. The filter is also not visible to the user anywhere.

```js
const view = {
  filters: [
    { field: '...', operator: '...', value: '...', isLocked: true }, //  Cannot be edited or removed by the user.
    { field: '...', operator: '...', value: '...' }, // Can be edited or removed by the user.
  ]
};
```

## Testing Instructions

- Go to the Pages page. 
- In any view other than "All", verify the "status" filter is not be visible anywhere (add filters button, column header, etc.).
- Test that the view works as it was not filtered: adding / removing other filters work as expected, etc.

## Props

Add props to @Souptik2001 for the original PR:

```
Co-authored-by: Souptik2001 <souptik@git.wordpress.org>
```
